src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/SqlProjectPublishService.cs

```diff
@@ -13,6 +13,29 @@ public async Task PublishSqlProject(IResourceWithDacpac resource, IResourceWithC
 
         try
         {
+            List<Task> waitingTasks = [];
+            if (target is SqlServerDatabaseResource)
+            {
+                waitingTasks.Add(resourceNotificationService.WaitForDependenciesAsync(target, cancellationToken));
+            }
+
+            if (resource.TryGetAnnotationsOfType<WaitAnnotation>(out var waitAnnotations))
+            {
+                foreach (var waitAnnotation in waitAnnotations)
+                {
+                    var targetState = waitAnnotation.WaitType switch
+                    {
+                        WaitType.WaitForCompletion => KnownResourceStates.Finished,
+                        WaitType.WaitUntilHealthy => KnownResourceStates.Running,
```

Comment:
Healthy != running.  Healthy is running AND health checks have passed.  There is a `WaitForResourceHealthy` method on resource notification service that can be used for this.